esp-wifi/src/wifi_interface.rs

```diff
@@ -221,6 +231,13 @@ impl<'a> WifiStack<'a> {
                     }
                 }
             }
+
+            if let (Some(dns_server), Some(dns_handle)) =
```

Comment:
this should be moved to the `smoltcp::socket::dhcpv4::Event::Configured(config)` case. Then we can get rid of `let mut dns_server = None;`